x/staking/keeper/delegation.go

```diff
@@ -626,6 +627,152 @@ func (k Keeper) Delegate(
 	return newShares, nil
 }
 
+// TransferDelegation changes the ownership of at most the desired number of shares.
+// Returns the actual number of shares transferred. Will also transfer redelegation
+// entries to ensure that all redelegations are matched by sufficient shares.
+// Note that no tokens are transferred to or from any pool or account, since no
+// delegation is actually changing state.
```

Comment:
My view has been that transfer delegation should be in the upstream staking module. It's useful in a lot of scenarios and I don't think it harms proof of stake security. I think there should be a message type for in the delegation module

Reply:
Agree, this is useful and should be upstreamed to the SDK

Reply:
Filed https://github.com/Agoric/agoric-sdk/issues/4315 for a later PR to surface this feature.